### PR TITLE
API docs updates

### PR DESCRIPTION
**Pull request by Yoshanuikabundi**
This PR:
- [x] Adds an improved stylesheet for the API reference
- [x] Fixes the root cause necessitating workarounds in `equil_rfe_methods.py` to document protocol settings (important modules were mocked when they were present in the RTD environment)
- [x] Organises the API reference with Autosummary for easier browsing
- [x] Adds more member documentation to the API docs
- [x] Sets up Intersphinx for linking to other documentation
- [x] Fixes some typos and mistakes in the installation and user guide docs

There's also one possibly controversial change:
- [x] Annotate `RelativeHybridTopologyProtocolSettings.forcefield_settings: OpenMMSystemGeneratorFFSettings`. Previously it inherited the `Settings.forcefield_settings: BaseForceFieldSettings` annotation. This is a Pydantic model so this annotation is a runtime behavior change.

@dwhswenson and I briefly discussed this today. It seems like it may just be an oversight; `RelativeHybridTopologyProtocol` certainly assumes that it will be `OpenMMSystemGeneratorFFSettings` and has no support for other classes, but maybe downstream consumers are re-using the settings class for their own protocols. The change makes it much easier to understand and find what's supposed to go where. I'm happy to revert this for now though and sort it out when I kick up a stink about the 3 different default `RelativeHybridTopologyProtocolSettings` values :sweat_smile:

There are new Sphinx warnings, but they relate to docstrings in GUFE that I'll submit a PR to fix in a moment - I'm pretty confident I know what's going on but if I'm wrong I can fix it here too.

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Comment by Yoshanuikabundi** [comment]
RTD is failing because it is configured not to permit warnings; the warnings introduced by this PR are fixed in https://github.com/OpenFreeEnergy/gufe/pull/210

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 1]
```diff
@@ -8,111 +8,17 @@
...
-        @classmethod
-        def get_defaults(cls):
-            return cls(
-                forcefield_settings=OpenMMSystemGeneratorFFSettings(),
-                thermo_settings=ThermoSettings(temperature=300 * unit.kelvin),
-            )
-else:
-    from gufe.settings import Settings, SettingsBaseModel  # type: ignore
-
+from gufe.settings import (
```
Comment (richardjgowers) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:16]: thanks for solving this, this was my nasty lack minute hack to get this working

File: docs/conf.py [hunk 2]
```diff
@@ -12,54 +12,76 @@
...
+    "sphinx.ext.autosectionlabel",
+    "sphinx_design",
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.autosummary",
+    "docs._ext.sass",
 ]
 
-autoclass_content = 'both'
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3.7", None),
```
[the remark this answers is not in the document]
Comment (Yoshanuikabundi) [reply docs/conf.py:46]: Do we follow [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html) (in which case 3.9)?

[the remark this answers is not in the document]
Comment (Yoshanuikabundi) [reply docs/conf.py:46]: The installation guide just says 3.9 and 3.10 are tested so I'll update this to 3.9.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py (hunk 1 above)
Comment (Yoshanuikabundi) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:16]: Yeah it was a pain to diagnose but ended up being extremely easy to fix :sweat_smile:

File: docs/_ext/sass.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Apache 2.0
+
+Modified to:
+- Write directly to Sphinx output directory
+- Infer targets if not given
+- Ensure ``target: Path`` in ``configure_path()``
+- Return version number and thread safety from ``setup()``
+- Use compressed style by default
+- More complete type checking
+"""
```
Comment (dwhswenson) [inline docs/_ext/sass.py:14]: Is there a reason not to open a PR upstream for these changes? I'm fine with vendoring for now, but if upstream will take the changes, I'd rather be able to remove this from our maintenance burden.

File: docs/guide/introduction.rst [hunk 4]
```diff
@@ -34,7 +34,7 @@ The output of **setup** is an :class:`.AlchemicalNetwork`. This contains all
 the information about what is being simulated (e.g., what ligands, host proteins, solvation details etc) and the
 information about how to perform the simulation (the Protocol).
 
-The output of the **executation** stage is the basic results from each edge.
+The output of the **execution** stage is the basic results from each edge.
```
Comment (dwhswenson) [inline docs/guide/introduction.rst:37]: 😂 This had to be me writing, but why was I repeatedly typing "executation"? Thanks for the typo fixes!

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 5]
```diff
@@ -519,19 +424,35 @@ class RelativeHybridTopologyProtocolSettings(Settings):
     class Config:
         arbitrary_types_allowed = True
 
+    # Inherited things
+
+    forcefield_settings: OpenMMSystemGeneratorFFSettings
+    """Parameters to set up the force field with OpenMM Force Fields."""
+    thermo_settings: ThermoSettings
+    """Settings for thermodynamic parameters."""
+
```
Comment (dwhswenson) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:433]: I take it there's no reasonable way to inherit these docstrings from parent classes? (Curious why, for my own understanding.)

**Review by dwhswenson — APPROVED** [review]
Looks good; some major improvements on what we had!

I had a couple of questions, but nothing that should block merging.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 6]
```diff
@@ -519,19 +424,35 @@ class RelativeHybridTopologyProtocolSettings(Settings):
...
     # Things for creating the systems
     system_settings: SystemSettings
+    """Simulation system settings including the long-range non-bonded method."""
     solvation_settings: SolvationSettings
+    """Settings for solvating the system."""
 
     # Alchemical settings
     alchemical_settings: AlchemicalSettings
+    """Alchemical protocol settings including lambda windows and soft cores."""
```
Comment (IAlibay) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:442]:
```suggestion
    """Alchemical protocol settings including lambda windows and soft core scaling choices."""
```
It's not a major change just a tiny clarification.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 7]
```diff
@@ -519,19 +424,35 @@ class RelativeHybridTopologyProtocolSettings(Settings):
     class Config:
         arbitrary_types_allowed = True
 
+    # Inherited things
+
+    forcefield_settings: OpenMMSystemGeneratorFFSettings
+    """Parameters to set up the force field with OpenMM Force Fields."""
```
[the remark this answers is not in the document]
Comment (IAlibay) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:430]: Discussed in today's call - there's probably nothing we can do about this for v1.0.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 8]
```diff
@@ -519,19 +424,35 @@ class RelativeHybridTopologyProtocolSettings(Settings):
...
     system_settings: SystemSettings
+    """Simulation system settings including the long-range non-bonded method."""
     solvation_settings: SolvationSettings
+    """Settings for solvating the system."""
 
     # Alchemical settings
     alchemical_settings: AlchemicalSettings
+    """Alchemical protocol settings including lambda windows and soft cores."""
     alchemical_sampler_settings: AlchemicalSamplerSettings
+    """Settings for sampling within lambda windows."""
```
Comment (IAlibay) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:444]:
```suggestion
    """
    Settings for alchemical space sampling, including the number of lambda windows
    and simultaneous sampling replicas.
    """
```
Something like that, simultaneous sampling replicas is quite a mouthful so feel free to reject. P.S. forgive the formatting it's late.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 9]
```diff
@@ -8,111 +8,17 @@
...
-            )
-else:
-    from gufe.settings import Settings, SettingsBaseModel  # type: ignore
-
+from gufe.settings import (
+    Settings,
+    SettingsBaseModel,
+    OpenMMSystemGeneratorFFSettings,
+    ThermoSettings,
+)
```
Comment (IAlibay) [inline openfe/protocols/openmm_rfe/equil_rfe_settings.py:22]:
```suggestion


```
Le PEP8?

**Review by IAlibay — COMMENTED** [review]
it's late, feel free to accept / reject as needed.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py [hunk 10]
```diff
@@ -149,7 +55,6 @@ def is_positive_distance(cls, v):
             raise ValueError(errmsg)
         return v
 
-
```
[the remark this answers is not in the document]
Comment (Yoshanuikabundi) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:152]: This is what happens when I can't use Black :(

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py (hunk 8 above)
Comment (Yoshanuikabundi) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:444]: The number of lambda windows is in `AlchemicalSettings`, and alchemical space sampling sounds to me like it would be searching through a space that includes different alchemical states. `AlchemicalSamplerSettings` seems focused on how to sample each window - could you clarify why my phrasing is incorrect/inadequate?

Comment (IAlibay) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:444]: re: number of lambda windows - you're 100% right, I'm muddling myself with n_replicas.

re: why I think we need to maybe reword - beyond "sampling within lambda space", these settings also tackle things like the online analysis interval, and (unfortunately) the number of whole repeats of the entire simulation. I was trying (and probably failing to) reword it as something that covers that wider space.

File: docs/_ext/sass.py (hunk 3 above)
Comment (Yoshanuikabundi) [reply docs/_ext/sass.py:14]: The crucial change is that it now writes to the build directory rather than the source directory, which is helpful for minimising the mess that Sphinx re-builds can get into. Once I vendored it to fix that, it seemed worthwhile to make more changes to make it a more natural fit. I'll open a PR upstream and see what happens :)

File: docs/guide/introduction.rst (hunk 4 above)
Comment (Yoshanuikabundi) [reply docs/guide/introduction.rst:37]: I was like, "is this a word with a slightly different meaning that no-one understands the distinction of", like dynamic vs dynamical or alchemic vs alchemical :laughing:

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py (hunk 5 above)
Comment (Yoshanuikabundi) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:433]: For `forcefield_settings`, the base class is meant to be generic over other force field generators. For `thermo_settings`, it could be placed in the base class with a PR to GUFE.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py (hunk 8 above)
Comment (Yoshanuikabundi) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py:446]: Gotcha, I misunderstood how the protocol worked.

File: openfe/protocols/openmm_rfe/equil_rfe_settings.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (Yoshanuikabundi) [reply openfe/protocols/openmm_rfe/equil_rfe_settings.py]: Sounds like a can of worms I probably shouldn't crack open just yet :laughing:

**Comment by Yoshanuikabundi** [comment]
Am I good to merge this?
